File: Java/Medium/WrongAnswer/count-the-number-of-beautiful-subarrays_9.java
class Solution {
    public long beautifulSubarrays(int[] nums) {
        
        int n = nums.length;
        int[] odd = new int[n + 1];
        int sum = 0;
        for (int i = 0; i < n; i++) {
            sum = sum + (nums[i] & 1);
            odd[sum]++;
        }
        long ans = 0;
        for (int i = 0; i <= n; i++) {
            ans += (long)odd[i] * (odd[i] - 1) / 2;
        }
        return ans;
    }
}

File: C++/Medium/WrongAnswer/count-the-number-of-beautiful-subarrays_9.cpp
class Solution {
public:
    long long beautifulSubarrays(vector<int>& nums) {
        
        long long res = 0;
        int n = nums.size();
        int cnt = 0;
        vector<int> v;
        for (int i = 0; i < n; ++i) {
            if (nums[i] % 2 == 1) {
                v.push_back(cnt);
                cnt = 0;
            } else {
                cnt++;
            }
        }
        v.push_back(cnt);
        for (int i = 1; i < v.size(); ++i) {
            res += (long long)(v[i] + 1) * (long long)(v[i - 1] + 1);
        }
        return res;
    }
};